lib/swm050/timer.c

```diff
@@ -0,0 +1,457 @@
+ * @param timer_int_en Passed to @ref timer_int_enable()
+ * @param edge_mode Passed to @ref timer_edge_mode()
+ * @param loop_mode Passed to @ref timer_loop_mode()
+ */
+void timer_pulse_capture_setup(uint32_t timer,
+			       bool timer_int_en,
+			       enum timer_edge_modes edge_mode,
+			       enum timer_loop_modes loop_mode)
+{
+	timer_setup_internal(timer, timer_int_en, TIMER_MODE_PULSE_CAPTURE,
+			     edge_mode, loop_mode, TIMER_CLK_INTERNAL,
+			     TIMER_OUTPUT_NONE, TIMER_LEVEL_LOW);
+}
+
+/**
+ * Setup the timer in duty cycle capture mode.
+ * @note Call @ref timer_enable() when you are ready to start the timer.
+ * @note Be sure to set the alternate functions of the timer pins
+ *	with @ref syscon_sel_af() and disable SWD on those pins
+ *	with @ref syscon_sel_swd() as needed.
+ * @note If interrupts are enabled here, the interrupt should also be enabled
+ *	using the NVIC before enabling the timer.
+ * @param timer Select timer @ref timer_select
+ * @param timer_int_en Passed to @ref timer_int_enable()
+ * @param edge_mode Passed to @ref timer_edge_mode()
+ * @param loop_mode Passed to @ref timer_loop_mode()
+ */
+void timer_duty_cycle_capture_setup(uint32_t timer,
+				    bool timer_int_en,
+				    enum timer_edge_modes edge_mode,
+				    enum timer_loop_modes loop_mode)
+{
+	timer_setup_internal(timer, timer_int_en, TIMER_MODE_DUTY_CYCLE_CAPTURE,
+			     edge_mode, loop_mode, TIMER_CLK_INTERNAL,
+			     TIMER_OUTPUT_NONE, TIMER_LEVEL_LOW);
+}
+
+/**
+ * Set the timer clock divider, based off of the 18MHz oscillator
+ * @param div Timer clock divider. Only the 6 least-significant bits are used,
+ *	Takes values from 0 to 63 (in reality the possible values are the even
+ *	numbers from 2 to 62, as well as the number 1). Anything after the 6
+ *	least-significant bits are stripped off of the value. If the value is 0,
+ *	it will be treated as a 1.  All odd values other than 1 are rounded down
+ *	to the closest even value, due to the fact that all odd values are
+ *	treated by the register as a 1, which would likely be unexpected. A
+ *	value of 0 would also normally be treated as a 2, which would also be
+ *	unexpected behavior.
+ */
+void timer_clock_div(uint8_t div)
+{
+	if (div <= 1) {
+		/* Make the value odd, meaning no divide. */
+		SYSCTL_SYS_CFG_1 |= (1 << 16);
+	} else {
+		/* Drop to the closest even value. */
+		SYSCTL_SYS_CFG_1 &= TIMER_DIV_MASK | ((~0x01 & div) << 16);
+	}
+}
+
+/**
+ * Enables or disables the timer.
+ * @param timer Select timer @ref timer_select
+ * @param en Enable or disable the timer
+ */
+void timer_enable(uint32_t timer, bool en)
+{
+	if (en) {
+		TIMER_CTRL(timer) |= TIMER_CTRL_EN;
+	} else {
+		TIMER_CTRL(timer) &= ~TIMER_CTRL_EN;
+	}
+}
+
+/**
+ * Enables or disables the timer's internal clock.
+ * @param timer Select timer @ref timer_select
+ * @param en Enable or disable the internal clock
+ */
+void timer_clock_enable(uint32_t timer, bool en)
+{
```

Comment:
traditionally, enabling clocks in a different peripheral would be the file for that peripheral, (ie, clk.c) not here, but I can see how much of a mess this target is, and you say this "finishes" the target, so it can just stay like this.